### DISCUSSION TITLE
Support sending attachments without a Content-Disposition header

### DISCUSSION THREAD

**Discussion by dcpurton** [neomutt/neomutt#3373] [Development]
I've recently noticed that most clients do not set `Content-Disposition: inline` for the main body of an email message. This header is left out. It's optional, so this is allowed.

NeoMutt *always* includes a `Content-Disposition` header no matter what. It can only be toggled between `inline` and `attachment`.

I've also discovered that some mail clients do not like this header to be present for the main body of the email. If it is, then the body part is displayed in a list of attachments (even for `Content-Disposition: inline`). This is not desirable.

NeoMutt supports the following enum for Content-Dispostion (from `email/mime.h`):

```C
enum ContentDisposition
{
  DISP_INLINE,    ///< Content is inline
  DISP_ATTACH,    ///< Content is attached
  DISP_FORM_DATA, ///< Content is form-data
  DISP_NONE,      ///< No preferred disposition
};
```

There is no user interface to select anything other than `DISP_INLINE` or `DISP_ATTACH`. But `DISP_NONE` is supported already when sending the email. The menu drawing functions also support `I`, `A`, `F`, `-` already too.

Things to wonder about:

- How might the interface for adjusting the Content-Disposition beyond just `inline` and `attachment` be adjusted in a backwards compatible way? (macros will expect a two way toggle for `^D` and this should not be changed)
- This also affects automatically created `multipart/mixed` attachments and `multipart/signed` attachments. These are always set to `DISP_INLINE`. Perhaps these should always be set to `DISP_NONE`?
- I haven't tested enough yet, but perhaps *all* multipart attachments should use `DISP_NONE` and not allow it to be changed. (This is enough to provide a working email in one client I checked)
- NeoMutt sets the disposition of an attachment to `DISP_INLINE` when it's parsed even if there is no `Content-Disposition` header. Should it instead use `DISP_NONE`?

**Comment by flatcap** [comment]
> It's optional

Ref: RFC2183 https://datatracker.ietf.org/doc/html/rfc2183

> RFC: and valid for any MIME entity ("message" or "body part")

> I've also discovered that some mail clients do not like this header to be present for the main body of the email

This is where things get tricky.

We want to be strict in what we send, but a little lax in what we accept.
We don't want to start making our code more complicated because other clients are broken.
